ios/BT/Extensions/Foundation/Date+Extensions.swift

```diff
@@ -2,4 +2,11 @@ extension Date {
   var posixRepresentation: Int {
     Int(timeIntervalSince1970) * 1000
   }
+
+  func difference(from date: Date, only component: Calendar.Component, calendar: Calendar = .current) -> Int {
```

Comment:
Should we be doing this calculation with a posix value?

Reply:
The API is a bit misleading, but it's neat - see `Persisted` - it call through to `setUserState`, so per your instinct these are all properties on a single model in the database (`UserState`)

Reply:
not in this case; we are comparing 2 `Date` objects (a `Date` stored in Realm and also a newly instantiated swift `Date`). Unless we are working with a date coming from the JS layer or a date that will eventually need to be sent up to the JS layer, we do not need to convert to a posix representation